pom.xml

```
@@ -232,6 +233,74 @@
         <module>process-springboot-example</module>
       </modules>
     </profile>
+
+    <profile>
+      <id>native</id>
+      <properties>
+        <failsafe.include>**/Native*IT.java</failsafe.include>
+        <failsafe.exclude></failsafe.exclude>
```

```suggestion
        <failsafe.include>**/*IT.java,**/IT*.java,**/*ITCase.java</failsafe.include>
        <failsafe.exclude></failsafe.exclude>
```

wouldn't it be simpler to restore it as default, since native test are anyway annotated?

---

Do you mean to not set the "failsafe.include" property (as the default one "**/*IT.java" already satisfy the native tests? If so, I'm fine with that, I will do it. But using the regular expressions that you suggest seems way more complex.

---

I meant to restore the Failsafe's default: https://maven.apache.org/surefire/maven-failsafe-plugin/examples/inclusion-exclusion.html#Inclusions

---

This won't work as Quarkus needs a special annotation @NativeImageTest which only has it the Native*IT.java classes to work. If a IT test is executed in native mode, it won't work.

---

> This won't work as Quarkus needs a special annotation @NativeImageTest which only has it the Native*IT.java classes to work

interesting, I didn't know that. Can you point me to the reference where the annotation enforces the name to be `Native*IT.java`, please?

---

I had a look to quarkus doc because I remember something similar but the required setup for native testing seems to be a bit different
https://quarkus.io/guides/getting-started-testing#native-executable-testing
https://quarkus.io/guides/building-native-image#testing-the-native-executable
I don't see any specific filter on resources

@Sgitario can you please verify?

---

Let me be clearer on this:

- The @NativeImageTest annotation from Quarkus does not require that the class name follows "Native*IT, we do. I meant that all the Native*IT.java classes we have in Kogito Examples, use this annotation (and that's why we are safe to use this regular expression in the pom.xml). 

- We need to set the property "failsafe.include" to "**/Native*IT.java" to not execute the *IT classes. Otherwise, when there are both *IT.java and Native*IT.java classes, the build fails at integration-test stage (the *IT test fails). You can try yourself.

However, let me know if you think something here can be improved or done differently.

---

Ok thanks for your explanation. I see your point, we have some ITs that requires native mode and others not so we decided to apply this naming convention to solve this.
I think we should better document it but it is out of scope for this PR :)